api/src/main/java/au/org/aodn/nrmn/restapi/controller/IngestionController.java

```diff
@@ -25,84 +26,157 @@
+            result.put("jobStatus", StatusJobType.FAILED);
+            result.put("message", "Job with given id has not been validated: " + jobId);
+            logger.warn(result.get("message").toString());
+            return ResponseEntity.badRequest().body(result);
+        }
 
-            surveyIngestionService.ingestTransaction(job, validatedRows);
+        if (!globalLockService.setLock()) {
+            result.put("jobStatus", StatusJobType.FAILED);
+            result.put("reason", "locked");
+            result.put("message", "Failed to set lock for job : " + jobId);
+            logger.warn(result.get("message").toString());
+            return ResponseEntity.badRequest().body(result);
+        }
 
-            materializedViewService.refreshAllAsync();
+        final var stagedJobLog = stagedJobLogRepository
+                .save(StagedJobLog.builder()
+                        .stagedJob(job)
+                        .eventType(StagedJobEventType.INGESTING)
+                        .build());
+
+        // Some upload is very large, use a separate to avoid timeout on web app.
+        logger.info("Ingest job id {} with transaction", job.getId());
+        executorService.execute(() -> {
+            try {
+                // Noted : use transaction template on executor task is needed to avoid no session error on
+                // Hiberate lazy load object happens after the main transaction completed inside call to
+                // ingestTransaction
+                tx.executeWithoutResult(s -> {
+                    logger.info("Execute ingest job id {} with transaction", job.getId());
+                    var rows = rowRepository.findRowsByJobId(job.getId());
+                    var species = speciesFormatting.getSpeciesForRows(rows);
+                    var validatedRows = speciesFormatting.formatRowsWithSpecies(rows, species);
+
+                    surveyIngestionService.ingestTransaction(job, validatedRows);
+
+                    logger.info("Refresh materialized view after job id {} ingested", job.getId());
+                    materializedViewService.refreshAllAsync();
+                });
+            }
+            catch (Exception e) {
+
+                logger.error("Ingestion Failed", e);
+
+                stagedJobLog.setDetails(e.getMessage());
+                stagedJobLog.setEventType(StagedJobEventType.ERROR);
+
+                stagedJobLogRepository.save(stagedJobLog);
+            }
+            finally {
+                globalLockService.releaseLock();
+            }
+        });
+
+        result.put("jobStatus", stagedJobLog.getEventType());
+        result.put("jobLogId", stagedJobLog.getId());
+        return ResponseEntity.ok(result);
+    }
 
-        } catch (Exception e) {
+    @GetMapping(path = "ingest/{job_id}")
+    @Operation(security = { @SecurityRequirement(name = "bearer-key") })
+    public ResponseEntity<Map> getIngest(@PathVariable("job_id") Long jobId) {
```

Comment:
`job_id` should be named `jobLogId` because this GET request takes `jobLogId` param instead of `jobId`